File: OS/2do Parcial/Project/Planning Procces/data/node.cpp
#include "node.hpp"

template <typename T>
Node<T>::Node(T datum)
{
    this->datum = datum;
    next = nullptr;
}

template <typename T>
Node<T>::~Node()
{
    delete this;
}

template <typename T>
string Node<T>::toString()
{
    return "Node{" + "datum=" + datum.toString() + ", next=" + next.toString() + "}";
}

File: OS/2do Parcial/Project/Modeling/Project/src/com/data/Node.java
package com.data;

public class Node <E>{
    private E datum;
    private Node next;

    public Node(E datum) {
        this.datum = datum;
        this.next = null;
    }

    public E getDatum() {
        return datum;
    }

    public Node getNext() {
        return next;
    }

    public void setNext(Node next) {
        this.next = next;
    }

    @Override
    public String toString() {
        return "Node{" +
                "datum=" + datum +
                ", next=" + next +
                '}';
    }
}
